src/main/frontend/handler/ui.cljs

```diff
@@ -301,5 +301,9 @@
     (state/pub-event! [:modal/show-cards])))
 
 (defn open-new-window!
-  []
-  (ipc/ipc "openNewWindow"))
+  ([_e]
+   (open-new-window! _e nil))
+  ([_e repo]
+   ; TODO: find out a better way to open a new window with a different repo path
```

Comment:
Better combine "openNewWindow" and "setCurrentGraph" and make a new ipc call.

Reply:
The set new repo url call is to persist the url into local storage. When logseq is opened in a new window, the app will be initialized with the repo url saved in the local storage.
You can see that there is no need to make an IPC to the node side. New logic added in this PR is within the browser window

Reply:
That's OK.